oauth2_provider/oauth2_validators.py

```diff
@@ -277,6 +277,11 @@ def _get_token_from_authentication_server(
             log.exception("Introspection: Failed POST to %r in token lookup", introspection_url)
             return None
 
+        if response.status_code == 403:
+            log.exception("Introspection: Failed to authenticate to %r",
```

Comment:
Use log.error(). You're not in an exception.

Reply:
sure